src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,3122 @@
+		} else {
+			stats.remove(AggregatorGroupControllingMetric.SCENE.getName() + AggregatorGroupControllingMetric.MESSAGE.getName());
+		}
+	}
+
+	/**
+	 * This method is used to populate polling interval
+	 *
+	 * @param stats store all statistics
+	 */
+	public void populatePollingInterval(Map<String, String> stats) {
+		Integer minPollingInterval = calculatingMinPollingInterval();
+
+		Long nextPollingInterval = System.currentTimeMillis() + localPollingInterval * 1000;
+		Date date = new Date(nextPollingInterval);
+		Format format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
+
+		stats.put(HubInfoMetric.MIN_POLLING_INTERVAL.getName(), minPollingInterval.toString());
+		stats.put(HubInfoMetric.NEXT_POLLING_INTERVAL.getName(), format.format(date));
+
+	}
+
+	//--------------------------------------------------------------------------------------------------------------------------------
+	//endregion
+
+	//region retrieve aggregated devices info: locations, aggregated device and device capabilities info in worker thread
+	//--------------------------------------------------------------------------------------------------------------------------------
+
+	/**
+	 * Retrieve aggregated devices and system information data -
+	 * and set next device/system collection iteration timestamp
+	 */
+	private void retrieveInfo() {
+		retrieveLocations(true);
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("New fetched location list: %s", cachedLocations));
+		}
+		retrieveRooms(true);
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("New fetched room list: %s", cachedRooms));
+		}
+		retrieveDevices(true);
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("New fetched device list: %s", cachedDevices));
+		}
+	}
+
+	/**
+	 * This method is used to retrieve list of device every 30 seconds by send GET request to https://api.smartthings.com/v1/devices?locationId={locationId}
+	 *
+	 * @param retryOnError retry if any error occurs
+	 * @throws ResourceNotReachableException When getting the empty response
+	 * @throws ResourceNotReachableException If any error occurs
+	 */
+	private void retrieveDevices(boolean retryOnError) {
+		try {
+			String request = SmartThingsURL.DEVICES
+					.concat(SmartThingsConstant.QUESTION_MARK)
+					.concat(SmartThingsURL.LOCATION_ID)
+					.concat(locationIdFiltered);
+
+			DeviceWrapper responseDeviceList = this.doGet(request, DeviceWrapper.class);
+
+			if (responseDeviceList.getDevices() != null) {
+				cachedDevices = (ConcurrentHashMap<String, Device>) responseDeviceList.getDevices().stream().collect(Collectors.toConcurrentMap(Device::getDeviceId, Function.identity()));
+			}
+		} catch (Exception e) {
+			if (retryOnError) {
+				retrieveDevices(false);
+			}
+			logger.error(String.format("Aggregated Device Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage()), e);
```

Comment:
maybe would be nice to sneak in locationId here, to see what request this was exactly right off the bat

Reply:
I fixed it